Frontend and Backend API SingIn and SignUp

rnavaneeth992 opened the pull request:
Issue: #98 

Added files

LifeAsDev commented on src/app/layout.js:
you delete the MuiThemeProvider

LifeAsDev replied:
and also why you comment the metadata?

LifeAsDev commented on src/components/right/RigthSignUp.jsx:
```diff
@@ -81,228 +105,219 @@ const RigthSignUp = () => {
   };
 
-  const [isMobile, setIsMobile] = useState(false);
+  const [isMobile, setIsMobile] = useState(window.innerWidth <= 767);
```
warning, window is not defined

LifeAsDev reviewed (commented):
review your code

rnavaneeth992 commented:
MuIthemeprovider pushed

Metadata is not working if i am working with react-query 

window warning is not coming for me, even though i added some validation please check it @LifeAsDev @akshayw1

LifeAsDev commented:
> MuIthemeprovider pushed
> Great
> Metadata is not working if i am working with react-query
> Can you explain me that issue?
> window warning is not coming for me, even though i added some validation please check it @LifeAsDev @akshayw1
yeah I know, but I fixed that warning, you have a old file solve that, just set it to false

rnavaneeth992 commented:
> > MuIthemeprovider pushed
> > Great
> > Metadata is not working if i am working with react-query
> > Can you explain me that issue?
> > window warning is not coming for me, even though i added some validation please check it @LifeAsDev @akshayw1
> > yeah I know, but I fixed that warning, you have a old file solve that, just set it to false (Oh Thanks)

Metadata if i add (it is telling you can't use in 'use client' if i remove and add then react-query I am unable to use it that's why i removed

LifeAsDev commented:
I dont notice you add 'use client' in layout, Dont add 'use client' in layout that is bad

LifeAsDev commented:
create a provider for query like the muithemeprovider and delete 'use client' of layout

rnavaneeth992 commented:
> create a provider for query like the muithemeprovider and delete 'use client' of layout

OH OK Bro got it I will do it and update you

LifeAsDev commented on src/components/right/RigthSignUp.jsx:
```diff
@@ -81,228 +105,231 @@ const RigthSignUp = () => {
+      return window.innerWidth <= 767;
+    }
+    return false;
+  });
```
adsad

rnavaneeth992 commented:
> create a provider for query like the muithemeprovider and delete 'use client' of layout

@LifeAsDev I tried even though some different errors are coming 

Uncaught Error: invariant expected app router to be mounted
    at useRouter (navigation.js:125:15)
    at HotReload (hot-reloader-client.js:355:46)
    at renderWithHooks (react-dom.development.js:11009:18)
    at mountIndeterminateComponent (react-dom.development.js:16761:13)
    at beginWork$1 (react-dom.development.js:18345:16)
    at beginWork (react-dom.development.js:26741:14)
    at performUnitOfWork (react-dom.development.js:25587:12)
    at workLoopSync (react-dom.development.js:25303:5)
    at renderRootSync (react-dom.development.js:25258:7)
    at recoverFromConcurrentError (react-dom.development.js:24475:20)
    at performConcurrentWorkOnRoot (react-dom.development.js:24420:26)
    at workLoop (scheduler.development.js:261:34)
    at flushWork (scheduler.development.js:230:14)
    at MessagePort.performWorkUntilDeadline (scheduler.development.js:534:21)
    
 Even though I tried like Muithemeprovider 
 
 Can you tell how to do it

LifeAsDev commented:
do a commit, I will clone your repo

rnavaneeth992 commented:
@LifeAsDev Done can you check please

LifeAsDev commented:
you forget to add the children prop

```

'use client'
import { QueryClient, QueryClientProvider } from 'react-query';
const queryClient = new QueryClient();

const QueryProvider = () => {
    return (
        <QueryClientProvider client={queryClient}>
        </QueryClientProvider>
    );
  };
  export default QueryProvider;

```

LifeAsDev commented:
Look the children prop here

```
const MuiThemeProvider = ({ children }) => {
  return <ThemeProvider theme={theme}>{children} </ThemeProvider>;
};
```

LifeAsDev commented:
also dont wrap html and body with the providers only the children

```
export default function RootLayout({ children }) {
  return (
    <html lang="en">
      <body className={inter.className}>
        <QueryClientProvider client={queryClient}>
          <AuthProvider>
            <MuiThemeProvider>{children}</MuiThemeProvider>
          </AuthProvider>
        </QueryClientProvider>
      </body>
    </html>
  );
}

```

LifeAsDev commented:
Also you did not import the QueryProvider that you created on layout instead you imported the QueryClientProvider

rnavaneeth992 commented:
@LifeAsDev I pushed changes now, Can you check

LifeAsDev commented:
> @LifeAsDev I pushed changes now, Can you check

yeah I see, is it working without errors?

rnavaneeth992 commented:
> > @LifeAsDev I pushed changes now, Can you check
> 
> yeah I see, is it working without errors?

Ya it is working 
<img width="960" alt="image" src="https://github.com/opencodeiiita/Hitch-Frontend/assets/80904345/ecdc2ecb-51ae-4f04-8b11-335fdc4e2b69">

rnavaneeth992 commented:
@LifeAsDev @akshayw1 Can you check and merge it please
